integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/LogicalRelationVisitor.java

```diff
@@ -119,7 +128,33 @@ public String apply() {
     // driver format looks like oracle:<drivertype>:<user>/<password>@<database>
     // whereas postgres, mysql, and sqlserver use the scheme://hostname:port/db format.
     String url = PlanUtils.sanitizeJdbcUrl(relation.jdbcOptions().url());
-    OpenLineage.DatasetFacets datasetFacet = PlanUtils.datasetFacet(relation.schema(), url);
+    OpenLineage.DatasetFacets datasetFacet = datasetFacet(relation.schema(), url);
     return Collections.singletonList(PlanUtils.getDataset(tableName, url, datasetFacet));
   }
+
+  private List<OpenLineage.Dataset> handleKafkaRelation(LogicalRelation x) {
+    KafkaRelation relation = (KafkaRelation) x.relation();
+    String server = "";
+    String topic = "";
+    try {
+      Field sourceOptionsField = x.relation().getClass().getDeclaredField("sourceOptions");
+      sourceOptionsField.setAccessible(true);
+      CaseInsensitiveMap sourceOptions = (CaseInsensitiveMap) sourceOptionsField.get(x.relation());
+      String servers =
+          (String) asJavaOptional(sourceOptions.get("kafka.bootstrap.servers")).orElse("");
+      topic =
```

Comment:
According to docs, there 3 possible option to read from kafka 
'subscribe' 'subscribePattern' and 'assign'
https://spark.apache.org/docs/3.0.0-preview/structured-streaming-kafka-integration.html